recipes/MEDIA/media_prepare.py

```diff
@@ -0,0 +1,1209 @@
+https://catalogue.elra.info/en-us/repository/browse/ELRA-E0024/
+https://drive.google.com/drive/u/1/folders/1z2zFZp3c0NYLFaUhhghhBakGcFdXVRyf
+See README.md for more info.
+
+Author
+------
+Gaelle Laperriere 2023
+"""
+
+import xml.dom.minidom as DOM
+from tqdm import tqdm
+import logging
+import subprocess
+import csv
+import os
+import glob
+import re
+
+logger = logging.getLogger(__name__)
+
+
+def prepare_media(
+    data_folder,
+    save_folder,
+    skip_wav=True,
+    method="slu",
+    task="full",
+    skip_prep=False,
+):
+    """
+    Prepares the csv files for the MEDIA dataset.
+    Both following repositories are nesseray for transcriptions
+    and annotations (S0272) and audio (E0024).
+    https://catalogue.elra.info/en-us/repository/browse/ELRA-S0272/
+    https://catalogue.elra.info/en-us/repository/browse/ELRA-E0024/
+
+    Arguments
+    ---------
+    data_folder: str
+        Path where folders S0272 and E0024 are stored.
+    save_folder: str
+        Path where the csvs and preprocessed wavs will be stored.
+    skip_wav: bool, optional
+        Skip the wav files storing if already done before.
+    method: str, optional
+        Used only for 'slu' task.
+        Either 'full' or 'relax'.
+        'full' Keep specifiers in concepts.
+        'relax' Remove specifiers from concepts.
+    task: str, optional
+        Either 'slu' or 'asr'.
+        'slu' Parse SLU data.
+        'asr' Parse ASR data.
+    skip_prep: bool, optional
+        If True, skip data preparation.
+    """
+
+    if skip_prep:
+        return
+
+    if not os.path.exists(save_folder):
+        os.makedirs(save_folder)
+    if not os.path.exists(save_folder + "/wav"):
+        os.makedirs(save_folder + "/wav")
+    if not os.path.exists(save_folder + "/csv"):
+        os.makedirs(save_folder + "/csv")
+
+    if skip(
+        save_folder + "/csv/train.csv",
+        save_folder + "/csv/dev.csv",
+        save_folder + "/csv/test.csv",
+    ):
+        logger.info("Csv files already exist, skipping data preparation!")
+        return
+
+    if task == "slu":
+        if method == "relax" or method == "full":
+            logger.info("Processing SLU " + method + " Media Dataset")
+        else:
+            raise ValueError("Parameter method must be 'full' or 'relax'")
+    elif task == "asr":
+        logger.info("Processing ASR Media Dataset")
+    else:
+        raise ValueError("Parameter task must be 'asr' or 'slu'")
+
+    xmls = {
+        "media_lot1.xml": "train",
+        "media_lot2.xml": "train",
+        "media_lot3.xml": "train",
+        "media_lot4.xml": "train",
+        "media_testHC.xml": "test",
+        "media_testHC_a_blanc.xml": "dev",
+    }
+
+    wav_paths = glob.glob(data_folder + "/S0272/**/*.wav", recursive=True)
+    channels, filenames = get_channels("./channels.csv")
+    unused_dialogs = get_unused_dialogs(data_folder)
+    concepts_full, concepts_relax = get_concepts_full_relax(
```

Comment:
This function should be optional too as the test2 is not mandatory for the recipe